p2p/protocol/circuitv2/relay/constraints.go

```diff
@@ -73,13 +73,10 @@ func (c *constraints) AddReservation(p peer.ID, a ma.Multiaddr) error {
 	}
 
 	var asnReservations []time.Time
-	var asn string
-	// Only public addresses have an ASN. Skip checking ASN for private addresses as
-	// initialising the ASN store is a costly operation. Skipping this check reduces a lot of
-	// flakiness in tests
-	if ip.To4() == nil && manet.IsPublicAddr(a) {
-		asn, _ = asnutil.Store.AsnForIPv6(ip)
-		if asn != "" {
+	var asn uint32
+	if ip.To4() == nil {
+		asn = asnutil.AsnForIPv6(ip)
+		if asn != 0 {
```

Comment:
Is it fine to check this for Private addresses now? Previously this was a slow operation causing flakiness in CI

Reply:
I removed all the initialization, it operate straight on binary data mmaped from in `.rodata` part of the executable, one lookup is now ~50ns on my computer.
Would be more if this part of the executable is cold (not in L2 and or the pages are still tripped), but never something significant (there are ~17 steps of binary search to do).